creator-node/src/diskManager.js

```diff
@@ -332,6 +336,137 @@ class DiskManager {
+        }
+      }
+
+      return subdirectories
+    } catch (e) {
+      genericLogger.error(
+        `Error in diskManager - listSubdirectoriesInFiles: ${e}`
+      )
+    }
+  }
+
+  // list all the CIDs in /file_storage/files/AqN
+  static async listNestedCIDsInFilePath(filesSubdirectory) {
+    const cids = []
+    // find files older than 3 days in filesSubdirectory (eg /file_storage/files/AqN)
+    try {
+      const stdout = await this._execShellCommand(
+        `find ${filesSubdirectory} -mtime +${DAYS_BEFORE_PRUNING_ORPHANED_CONTENT}`
+      )
+
+      for (const file of stdout.split('\n')) {
+        const fileTrimmed = file.trim()
+        // if fileTrimmed is a non-null string and is not just equal to base directory
+        if (fileTrimmed && fileTrimmed !== filesSubdirectory) {
+          const parts = fileTrimmed.split('/')
+          cids.push(parts[parts.length - 1])
+        }
+      }
+
+      return cids
+    } catch (e) {
+      genericLogger.error(
+        `Error in diskManager - listNestedCIDsInFilePath: ${e}`
+      )
+    }
+  }
+
+  static async sweepSubdirectoriesInFiles() {
+    const subdirectories = await this.listSubdirectoriesInFiles()
+    if (!subdirectories) return
+
+    for (let i = 0; i < subdirectories.length; i += 1) {
+      try {
+        const subdirectory = subdirectories[i]
+        genericLogger.info(
+          `diskManager#sweepSubdirectoriesInFiles - iteration ${i} out of ${subdirectories.length}`
+        )
+        const cids = await this.listNestedCIDsInFilePath(subdirectory)
+
+        // TODO - parallelize into batches
+        // const cidsArray = await Promise.all(subdirectories.slice(i, i+5).map(s => listNestedCIDsInFilePath(s)))
+        // const cids = cidsArray.flat()
+
+        const queryResults = await models.File.findAll({
+          attributes: ['multihash', 'storagePath'],
+          raw: true,
+          where: {
+            multihash: {
+              [models.Sequelize.Op.in]: cids
+            }
+          }
+        })
+
+        const cidToFileLookup = {}
+        for (const file of queryResults) {
+          cidToFileLookup[file.multihash] = file.storagePath
+        }
+
+        const cidsToDelete = []
+        const cidsNotToDelete = []
+        for (const cid of cids) {
+          // if db doesn't contain file, log as okay to delete
+          if (!cidToFileLookup.hasOwnProperty(cid)) {
+            cidsToDelete.push(cid)
+          } else cidsNotToDelete.push(cid)
+        }
+
+        if (cidsNotToDelete.length > 0) {
+          genericLogger.info(
+            `diskmanager.js - not safe to delete ${cidsNotToDelete.toString()}`
+          )
+        }
+
+        if (cidsToDelete.length > 0) {
+          genericLogger.info(
+            `diskmanager.js - safe to delete ${cidsToDelete.toString()}`
+          )
+
+          // gate deleting files on disk with the same env var
+          if (config.get('syncForceWipeEnabled')) {
```

Comment:
nice so is this tested with the env var disabled and just spot checking the CIDs it logs out? seems like a reliable way to dry run